facilitator/src/bin/facilitator.rs

```diff
@@ -617,7 +616,28 @@ fn main() -> Result<(), anyhow::Error> {
         ("aggregate", Some(sub_matches)) => aggregate(sub_matches),
         ("lint-manifest", Some(sub_matches)) => lint_manifest(sub_matches),
         (_, _) => Ok(()),
+    };
+
+    // Once we've run the subcommand, whether it succeeds or fails, we want to push any metrics
+    // we have collected.
+    if let Some(pushgateway) = matches.value_of("pushgateway") {
+        if pushgateway != "" {
```

Comment:
nit: [You can add a condition to a pattern match](https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html#extra-conditionals-with-match-guards) so that it'll match `Some` and check for the empty string. I'm not sure if it'll work in a `let` statement though.

Reply:
Yeah, that was my first instinct, but I decided

```
if let Some(pushgateway) if !pushgateway.is_empty() = matches.value_of("pushgateway") {
```

was stylistically too confusing. Happy to go the other way on this though; what do you think?

Reply:
I prefer doing it in a single statement to avoid the extra indentation of the nested if statement, and I think statements that you have to squint at a bit to make sense of are table stakes when you write Rust.

Reply:
Oh, funny, I actually looked this up the other day and found it's not yet implemented, then promptly forgot: https://github.com/rust-lang/rust/issues/51114

So, guard statements (the "if !pushgateway.is_empty()") are not yet supported on `if let`.

Reply:
Shame! in that case what you have is just fine.